### PR TITLE
[cleanup] Remove NULLs in boolean conditions 

### PR DESCRIPTION
**Pull request by thexai**
## Description
Remove NULLs in boolean conditions 

In general is replaced:
- `if (x != NULL)`  --> `if (x)` 
- `if (x == NULL)`  --> `if (!x)`

Automated replaced using reg. expressions with Notepad++.

## Motivation and context
Remove 1275 NULLs  😃

In some cases `NULL` are replaced by `nullptr` so that the code is better understood. 

e.g.

`return (pfile);`  may be confusing without knowing if the function returns bool then is used `return (pfile != nullptr);` in that cases.

## How has this been tested?
Build Windows x64 and runtime quick test.
Reviewed manually although something may have escaped me. It has been quite complex due to format differences in spaces and unusual things like `if (NULL!=pfile)`.

## What is the effect on users?
Nothing

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [X] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [X] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/Autorun.cpp [hunk 1]
```diff
@@ -77,7 +77,7 @@ void CAutorun::ExecuteAutorun(const std::string& path, bool bypassSettings, bool
 
   CCdInfo* pInfo = CServiceBroker::GetMediaManager().GetCdInfo(path);
 
-  if ( pInfo == NULL )
+  if ( !pInfo )
```
Comment (enen92) [inline xbmc/Autorun.cpp:80]: Additional spaces after parenthesis. Clang-format should find such stuff isn't it?

Comment (thexai) [reply xbmc/Autorun.cpp:80]: Yep I will apply jenkins diff separated in each commit for that. Also some code that occupies two lines can now be fitted into a single line.

**Comment by phunkyfish** [comment]
Be careful with this PR. Things can be quirky with how C based char functions work. I don't recall the example I came across before but it was a pain to find afterwards.

**Comment by thexai** [comment]
@phunkyfish 
Maybe you are thinking on changing variable assignment/initialization from NULL to nullptr. This is more dangerous because is need to control all places on that variable is used.

In this PR is none variable assignment has changed. Only bool conditions style. Is still possible introduce bugs because some typo but in general all changes are trivial.

**Comment by howie-f** [comment]
hm, my point on this is readability: example:
https://github.com/xbmc/xbmc/blob/b50817d1622a410a6345e2f28c4ff976109751d7/xbmc/LangInfo.cpp#L396-L402
at a first glance no idea..... `m_languageAddon` is that a bool, is that a pointer?
yes you can dig that out but a `if (m_languageAddon == nullptr)` is easier imo.
appreciate the effort but i'm split :/

**Comment by phunkyfish** [comment]
> @phunkyfish Maybe you are thinking on changing variable assignment/initialization from NULL to nullptr. This is more dangerous because is need to control all places on that variable is used.
> 
> In this PR is none variable assignment has changed. Only bool conditions style. Is still possible introduce bugs because some typo but in general all changes are trivial.

Honestly I don’t remember, it was something to do with NULL being equal to 0 but nullptr is not equal to zero and some cstring functions. The problem was the comparison but for the life of me I don’t remember the example.

File: xbmc/Autorun.cpp (hunk 1 above)
Comment (garbear) [reply xbmc/Autorun.cpp]: You can also do post-commit hooks. A team member pointed me to https://github.com/barisione/clang-format-hooks.

**Comment by garbear** [comment]
There's some varying opinion among team members about converting raw pointers to bool (but we universally agree shared_ptr should always use `!` like in your PR).

Personally, I think that type context is important, so to cast from pointer to bool I'd do `if (ptr != nulltr)` instead of `if (!ptr)` because `!` is a logical bool-to-bool operator. The comparison to `nullptr`, on the other hand, is more type-correct because `!=` compares two pointers and returns a bool.

**Comment by garbear** [comment]
Also cleanup like this is appreciated!

**Pull request closed without merging**